### PR TITLE
feat: add unit tests presentation

### PR DESCRIPTION
**Pull request by PrykhodzkaV**

### CODE REVIEW

File: unit-testing/presentation.html [hunk 1]
```diff
@@ -1,27 +1,310 @@
...
+          <li class="fragment">Curtail maintain costs</li>
+          <li class="fragment">Improve time-to-market</li>
+          <li class="fragment">Enhance quality of apllication</li>
+          <li class="fragment">Boost competitiveness</li>
+        </ul>
+      </section>
+      <!-- 4 slide -->
+      <section>
+        <h3 class="slide-name">The scope of Testing​</h3>
+        <p class="font-size-2 align-center">
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:63]: could you place it in the center vertically

File: unit-testing/presentation.html [hunk 2]
```diff
@@ -1,27 +1,310 @@
...
+        </code></pre>
+      </section>
+      <!-- 20 slide -->
+      <section>
+        <!-- 20/0 slide -->
+        <section>
+          <h3 class="slide-name">Test coverage​</h3>
+          <p class="font-size-2 align-center">With that being said it is generally accepted that 80% coverage is a good
+            goal to aim for.</p>
+          <p class="font-size-2 align-center">Trying to reach a higher coverage might turn out to be costly, while not
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:275]: could you explain in more details it.
probably provide example what should not be covered

File: unit-testing/presentation.html [hunk 3]
```diff
@@ -1,27 +1,310 @@
...
+      </section>
+      <!-- 21 slide -->
+      <section>
+        <h3 class="slide-name">When tests fail​​</h3>
+        <img src='assets/images/failed_tests.png' height="500"/>
+      </section>
+      <!-- 22 slide -->
+      <section>
+        <h3 class="slide-name">Husky</h3>
+        <pre><code data-trim data-noescape>
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:299]: could you describe in few words what it is and benefits

File: unit-testing/presentation.html [hunk 4]
```diff
@@ -1,27 +1,310 @@
...
+      "husky": {​​
+        "hooks": {​​
+          "pre-commit": "YOUR_COMMAND",​​
+          "pre-push": "YOUR_COMMAND"​​
+        }​​
+      }​​
+    </code></pre>
+      </section>
     </div>
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:308]: please add QA slide

File: unit-testing/presentation.html [hunk 5]
```diff
@@ -1,27 +1,310 @@
...
+          jest.spyOn(object, methodName)​​
+          jest.clearAllMocks()​​
+          jest.resetAllMocks()​​
+          jest.restoreAllMocks()​​
+        </code></pre>
+      </section>
+      <!-- 19 slide -->
+      <section>
+        <h3 class="slide-name">Snapshots​​</h3>
+        <pre><code data-trim data-noescape>
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:261]: could you add proc / cons of this approach?

File: unit-testing/presentation.html [hunk 6]
```diff
@@ -1,27 +1,310 @@
...
+          .toContain(item)​​
+          .toEqual(value)​​
+          .toMatch(regexp | string)​​
+          .toMatchSnapshot(propertyMatchers?, hint?)​​
+          .toThrow(error?)​​
+        </code></pre>
+      </section>
+      <!-- 18 slide -->
+      <section>
+        <h3 class="slide-name">Mocking & The Jest Object​</h3>
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:242]: could you add few words about why we need to use mocks?
stubs VS mocks?

File: unit-testing/presentation.html [hunk 7]
```diff
@@ -1,27 +1,310 @@
...
+          <li class="fragment">The risks of simple mistakes leaking to the end user significantly increase in the
+            absence of unit tests.</li>
+          <li class="fragment">Skipping unit tests quickly turns fresh code into legacy code, which is harder to
+            maintain.</li>
+        </ul>
+      </section>
+      <!-- 8 slide -->
+      <section>
+        <h3 class="slide-name">TDD vs BDD</h3>
+        <img src='assets/images/tdd_bdd.png' height="500"/>
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:96]: I'm not really understand this picture

File: unit-testing/presentation.html [hunk 8]
```diff
@@ -1,27 +1,310 @@
...
+      </section>
+      <!-- 9 slide -->
+      <section>
+        <h3 class="slide-name">TDD is RED-GREEN-BLUE development​</h3>
+        <img src='assets/images/tdd.png' height="500"/>
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:101]: could you try to remove height here and change in the common.css
```
img {
  - max-width: 100%;
  - max-height: 100%;
  + max-width: 65% !important;
  + max-height: 65% !important;
  object-fit: contain;
}```

File: unit-testing/presentation.html [hunk 9]
```diff
@@ -1,27 +1,310 @@
...
+  <link rel="stylesheet" href="../reveal-js/dist/theme/white.css" id="theme" />
+  <!-- Theme used for syntax highlighting of code -->
+  <link rel="stylesheet" href="../reveal-js/plugin/highlight/monokai.css" />
+  <link rel="stylesheet" href="../reveal-js/common.css" />
+</head>
+<body>
+  <div class="reveal">
+    <div class="slides">
+      <!-- 0 slide -->
+      <section class="center">
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:21]: please use
```
<section class="flex-container height-100">
          <h4 class="flex-item-horizontal-center font-size-3">
            Presentation Name
          </h4>
        </section>
```

File: unit-testing/presentation.html [hunk 10]
```diff
@@ -1,27 +1,310 @@
...
-    <div class="reveal">
-      <!-- Any section element inside of this container is displayed as a slide -->
-      <div class="slides">
-        <section class="center">
-          <h2 class="custom-presentation-name"> Presentation​ Name </h3>
-        </section>
-      </div>
+<head>
+  <meta charset="utf-8" />
+  <title>react-native-course - name module</title>
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:5]: react-native-course - unit-testing

**Review by Dzmitry-Klokau — CHANGES_REQUESTED** [review]

File: unit-testing/presentation.html (hunk 2 above)
Comment (PrykhodzkaV) [reply unit-testing/presentation.html]: done

File: unit-testing/presentation.html (hunk 3 above)
Comment (PrykhodzkaV) [reply unit-testing/presentation.html:401]: done

File: unit-testing/presentation.html (hunk 5 above)
Comment (PrykhodzkaV) [reply unit-testing/presentation.html]: done

File: unit-testing/presentation.html (hunk 6 above)
Comment (PrykhodzkaV) [reply unit-testing/presentation.html]: done

File: unit-testing/presentation.html (hunk 7 above)
Comment (PrykhodzkaV) [reply unit-testing/presentation.html]: done

File: unit-testing/presentation.html (hunk 10 above)
Comment (PrykhodzkaV) [reply unit-testing/presentation.html]: done

File: unit-testing/presentation.html [hunk 11]
```diff
@@ -31,14 +402,15 @@ <h2 class="custom-presentation-name"> Presentation​ Name </h3>
     <script src="../reveal-js/plugin/highlight/highlight.js"></script>
     <script>
       var url = new URL(window.location.href);
-      var showNotes = true; // TODO Uncomment before merging!!! url.searchParams.has("showNotes");
+      var showNotes = url.searchParams.has("showNotes");
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:406]: please add:

```
if (showNotes) {
      var head = document.getElementsByTagName("HEAD")[0];
      var link = document.createElement("link");
      link.rel = "stylesheet";
      link.type = "text/css";
      link.href = "../reveal-js/speaker-notes-mobile.css";
      head.appendChild(link);
}
```

File: unit-testing/presentation.html [hunk 12]
```diff
@@ -1,27 +1,398 @@
...
+</head>
+
+<body>
+  <div class="reveal">
+    <div class="slides">
+      <section class="presentation-name-slide">
+        <h4>Unit tests</h4>
+      </section>
+      <!-- 1 slide -->
+      <section>
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:27]: class="agenda-slide"

File: unit-testing/presentation.html [hunk 13]
```diff
@@ -1,27 +1,398 @@
...
+          <li class="fragment">Unit tests</li>
+          <li class="fragment">TDD & BDD</li>
+          <li class="fragment">Testing statistic</li>
+          <li class="fragment">Testing libraries</li>
+          <li class="fragment">Jest as a testing framework</li>
+          <li class="fragment">Husky</li>
+        </ul>
+      </section>
+      <!-- 2 slide -->
+      <section>
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:41]: class="bullets-slide"

File: unit-testing/presentation.html [hunk 14]
```diff
@@ -1,27 +1,398 @@
...
+        <h3 class="slide-name">Testing role in Software Development​</h3>
+        <ul>
+          <li class="fragment">Testing is about verifying that what was specified is what was delivered.</li>
+          <li class="fragment">The testing is important since it discovers defects/bugs before the delivery to the
+            client, which guarantees the quality of the software.​</li>
+        </ul>
+      </section>
+      <!-- 3 slide -->
+      <section>
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:50]: class="bullets-slide"

File: unit-testing/presentation.html [hunk 15]
```diff
@@ -1,27 +1,398 @@
...
+          <li class="fragment">Prevent targeted malware attacks</li>
+          <li class="fragment">Improves the user experience</li>
+          <li class="fragment">Curtail maintain costs</li>
+          <li class="fragment">Improve time-to-market</li>
+          <li class="fragment">Enhance quality of apllication</li>
+          <li class="fragment">Boost competitiveness</li>
+        </ul>
+      </section>
+      <!-- 4 slide -->
+      <section>
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:63]: class="bullets-slide"

File: unit-testing/presentation.html [hunk 16]
```diff
@@ -1,27 +1,398 @@
...
+        <h3 class="slide-name">The scope of Testing​</h3>
+        <ul>
+          <li class="fragment"> The test scope defines what areas, features, and functionalities of an application
+            should be tested</li>
+        </ul>
+      </section>
+      <!-- 5 slide -->
+      <section>
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:71]: class="picture-slide" and remove image height attribute

File: unit-testing/presentation.html [hunk 17]
```diff
@@ -1,27 +1,398 @@
...
+        <h3 class="slide-name">Testing Pyramid​​</h3>
+        <img src='assets/images/testing_pyramid.svg' height="500" />
+      </section>
+      <!-- 6 slide -->
+      <section>
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:76]: class="bullets-slide"

File: unit-testing/presentation.html [hunk 18]
```diff
@@ -1,27 +1,398 @@
...
+        <h3 class="slide-name">Benefits of unit tests​​</h3>
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:77]: please use h4 for all slide-names

File: unit-testing/presentation.html [hunk 19]
```diff
@@ -1,27 +1,398 @@
...
+          <li class="fragment">Save development time</li>
+          <li class="fragment">Provide quick access to code documentation</li>
+          <li class="fragment">Contribute to code quality​</li>
+          <li class="fragment">Improve application architecture​</li>
+          <li class="fragment">Debug process​</li>
+          <li class="fragment">Reduce costs in long run</li>
+        </ul>
+      </section>
+      <!-- 7 slide -->
+      <section>
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:88]: class="bullets-slide"

File: unit-testing/presentation.html [hunk 20]
```diff
@@ -1,27 +1,398 @@
...
+                </td>
+                <td>
+                  <p>Collaboration is required by all stakeholders in BDD</p>
+                </td>
+              </tr>
+            </tbody>
+          </table>
+      </section>
+      <!-- 9 slide -->
+      <section>
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:162]: class="picture-slide"  and remove image height attribute

File: unit-testing/presentation.html [hunk 21]
```diff
@@ -1,27 +1,398 @@
...
+        <h3 class="slide-name">TDD is RED-GREEN-BLUE development​</h3>
+        <img src='assets/images/tdd.png' height="500" />
+      </section>
+      <!-- 10 slide -->
+      <section>
+        <!-- 10/0 slide -->
+        <section>
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:169]: class="picture-slide"

File: unit-testing/presentation.html [hunk 22]
```diff
@@ -1,27 +1,398 @@
...
+          <h3 class="slide-name">Testing statistic​</h3>
+          <img src='assets/images/statistic_1.png' />
+        </section>
+        <!-- 10/1 slide -->
+        <section>
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:174]: class="picture-slide"

File: unit-testing/presentation.html [hunk 23]
```diff
@@ -1,27 +1,398 @@
...
+          <h3 class="slide-name">Testing statistic​</h3>
+          <img src='assets/images/statistic_2.png' />
+        </section>
+      </section>
+      <!-- 11 slide -->
+      <section>
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:180]: class="picture-with-description-slide" and please change order of text and image

File: unit-testing/presentation.html [hunk 24]
```diff
@@ -1,27 +1,398 @@
...
+        <h3 class="slide-name">Writing testable code​​</h3>
+        <p class="font-size-2 align-center">
+          To make your app more testable, start by separating the view part of your app from your business logic and app
+          state.</p>
+        <img src='assets/images/separate_logic_and_ui.png' />
+      </section>
+      <!-- 12 slide -->
+      <section>
+        <!-- 12/0 slide -->
+        <section>
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:190]: class="picture-slide" and change image alt

File: unit-testing/presentation.html [hunk 25]
```diff
@@ -1,27 +1,398 @@
...
+      <!-- 13 slide -->
+      <section>
+        <h3 class="slide-name">Why Jest?</h3>
+        <ul>
+          <li class="fragment">Facebook developed the Jest unit testing framework.</li>
+          <li class="fragment">The default template of React Native ships with Jest testing framework.</li>
+        </ul>
+      </section>
+      <!-- 14 slide -->
+      <section>
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:208]: class="text-and-conding-slide"

File: unit-testing/presentation.html [hunk 26]
```diff
@@ -1,27 +1,398 @@
...
+              so your tests are more focused and simpler in what they are trying to prove.</li>
+            <li class="fragment">Stub is for replacing a method with code that returns a specified result: initialize ->
+              exercise -> verify.</li>
+            <li class="fragment">Mock is a stub with an assertion that the method gets called: initialize -> set
+              expectations -> exercise -> verify.</li>
+            <li class="fragment">Stubs don't fail your tests, mock can.</li>
+          </ul>
+        </section>
+        <!-- 18/1 slide -->
+        <section>
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:296]: class="bullets-slide"

File: unit-testing/presentation.html [hunk 27]
```diff
@@ -1,27 +1,398 @@
...
+          jest.fn(implementation?)​​
+          jest.spyOn(object, methodName)​​
+          jest.clearAllMocks()​​
+          jest.resetAllMocks()​​
+          jest.restoreAllMocks()​​
+        </code></pre>
+        </section>
+      </section>
+      <!-- 19 slide -->
+      <section>
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:320]: class="bullets-slide"

File: unit-testing/presentation.html [hunk 28]
```diff
@@ -1,27 +1,398 @@
...
+        <pre class="fragment"><code data-trim>
+          it('should match snapshot', () => {
+            const {toJSON} = render(<MyButton title="title"></MyButton>);
+            expect(toJSON()).toMatchSnapshot();
+          });</code></pre>
+      </section>
+      <!-- 20 slide -->
+      <section>
+        <!-- 20/0 slide -->
+        <section>
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:343]: class="bullets-slide"

File: unit-testing/presentation.html [hunk 29]
```diff
@@ -1,27 +1,398 @@
...
+            <li class="fragment">100% unit test coverage does not mean we had good tests, or even that the tests are
+              complete.</li>
+            <li class="fragment">The tests could be missing important data and only testing with data that succeeds,
+              failing to test data that causes failures.</li>
+            <li class="fragment">Merely executing the code without checking its functionality still counts in the
+              coverage metrics.</li>
+          </ul>
+        </section>
+        <!-- 20/1 slide -->
+        <section>
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:359]: class="picture-slide", add header and remove image height attribute

File: unit-testing/presentation.html [hunk 30]
```diff
@@ -1,27 +1,398 @@
...
+          <img src='assets/images/coverage_1.png' height="500" />
+        </section>
+        <!-- 20/2 slide -->
+        <section>
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:363]: class="picture-slide", add header and remove image height attribute

File: unit-testing/presentation.html [hunk 31]
```diff
@@ -1,27 +1,398 @@
...
+          <img src='assets/images/coverage_2.png' height="500" />
+        </section>
+        <!-- 20/3 slide -->
+        <section>
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:367]: class="picture-slide", add header and remove image height attribute

File: unit-testing/presentation.html [hunk 32]
```diff
@@ -1,27 +1,398 @@
...
+          <img src='assets/images/coverage_3.png' height="500" />
+        </section>
+      </section>
+      <!-- 21 slide -->
+      <section>
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:372]: class="picture-slide", add header and remove image height attribute

File: unit-testing/presentation.html [hunk 33]
```diff
@@ -1,27 +1,398 @@
...
+        <h3 class="slide-name">When tests fail​​</h3>
+        <img src='assets/images/failed_tests.png' height="500" />
+      </section>
+      <!-- 22 slide -->
+      <section>
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:377]: class="text-and-conding-slide" and merge to p-s to one

File: unit-testing/presentation.html [hunk 34]
```diff
@@ -1,27 +1,398 @@
...
+        "pre-commit": "YOUR_COMMAND",​​
+        "pre-push": "YOUR_COMMAND"​​
+      }​​
+    }​​
+  </code></pre>
+      </section>
+      <section class="qa-slide">
+        <h4>QA</h4>
+      </section>
     </div>
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:396]: please add one more </div>

File: unit-testing/presentation.html [hunk 35]
```diff
@@ -1,27 +1,398 @@
...
+          <li class="fragment">The risks of simple mistakes leaking to the end user significantly increase in the
+            absence of unit tests.</li>
+          <li class="fragment">Skipping unit tests quickly turns fresh code into legacy code, which is harder to
+            maintain.</li>
+        </ul>
+      </section>
+      <!-- 8 slide -->
+      <section class="table-slide">
+        <h3 class="slide-name">TDD VS BDD</h3>
+        <table class="fragment" data-fragment-index="2">
```
Comment (Dzmitry-Klokau) [inline unit-testing/presentation.html:100]: you have two tables here..
please find and fix all issues it should help to fix issue connected with safari

**Review by Dzmitry-Klokau — CHANGES_REQUESTED** [review]

File: unit-testing/presentation.html (hunk 12 above)
Comment (PrykhodzkaV) [reply unit-testing/presentation.html]: done1

File: unit-testing/presentation.html (hunk 13 above)
Comment (PrykhodzkaV) [reply unit-testing/presentation.html]: done1

File: unit-testing/presentation.html (hunk 14 above)
Comment (PrykhodzkaV) [reply unit-testing/presentation.html]: done1

File: unit-testing/presentation.html (hunk 15 above)
Comment (PrykhodzkaV) [reply unit-testing/presentation.html]: done1

File: unit-testing/presentation.html (hunk 16 above)
Comment (PrykhodzkaV) [reply unit-testing/presentation.html]: done1

File: unit-testing/presentation.html (hunk 31 above)
Comment (PrykhodzkaV) [reply unit-testing/presentation.html]: done1

File: unit-testing/presentation.html (hunk 33 above)
Comment (PrykhodzkaV) [reply unit-testing/presentation.html]: done1

File: unit-testing/presentation.html (hunk 34 above)
Comment (PrykhodzkaV) [reply unit-testing/presentation.html:395]: done1

File: unit-testing/presentation.html (hunk 35 above)
Comment (PrykhodzkaV) [reply unit-testing/presentation.html:101]: done1, but didn't help to fix Safari

File: unit-testing/presentation.html (hunk 32 above)
Comment (PrykhodzkaV) [reply unit-testing/presentation.html]: done1

File: unit-testing/presentation.html (hunk 27 above)
Comment (Dzmitry-Klokau) [reply unit-testing/presentation.html:319]: ?

File: unit-testing/presentation.html (hunk 26 above)
Comment (Dzmitry-Klokau) [reply unit-testing/presentation.html:295]: ?

**Review by Dzmitry-Klokau — APPROVED** [review]

**Review by annakosakova — APPROVED** [review]

**Comment by unfalse** [comment]
I can't see any notes and the sidebar isn't opened in presentation. Please add notes for each slide.

File: unit-testing/presentation.html [hunk 36]
```diff
@@ -2,43 +2,512 @@
...
+          <h4 class="slide-name">Testing role in Software Development​</h4>
+          <ul>
+            <li class="fragment">
+              Testing is about verifying that what was specified is what was
+              delivered.
+            </li>
+            <li class="fragment">
+              The testing is important since it discovers defects/bugs before
+              the delivery to the client, which guarantees the quality of the
+              software.​
```
Comment (unfalse) [inline unit-testing/presentation.html:74]: The good presentation doesn't show big amounts of text on the slides. It must be in notes.

File: unit-testing/presentation.html [hunk 37]
```diff
@@ -2,43 +2,512 @@
...
+            <li class="fragment">Boost competitiveness</li>
+          </ul>
+        </section>
+        <!-- 4 slide -->
+        <section class="bullets-slide">
+          <h4 class="slide-name">The scope of Testing​</h4>
+          <ul>
+            <li class="fragment">
+              The test scope defines what areas, features, and functionalities
+              of an application should be tested
```
Comment (unfalse) [inline unit-testing/presentation.html:97]: This text is too long and doesn't really shows anything. It must be in notes. This slide need pictures and a list with items.

File: unit-testing/presentation.html [hunk 38]
```diff
@@ -2,43 +2,512 @@
...
+            <li class="fragment">Reduce costs in long run</li>
+          </ul>
+        </section>
+        <!-- 7 slide -->
+        <section class="bullets-slide">
+          <h4 class="slide-name">Impacts of neglecting unit tests</h4>
+          <ul>
+            <li class="fragment">
+              The risks of simple mistakes leaking to the end user significantly
+              increase in the absence of unit tests.
```
Comment (unfalse) [inline unit-testing/presentation.html:124]: Another boring slide with long texts.

File: unit-testing/presentation.html [hunk 39]
```diff
@@ -2,43 +2,512 @@
...
+            </li>
+            <li class="fragment">
+              Skipping unit tests quickly turns fresh code into legacy code,
+              which is harder to maintain.
+            </li>
+          </ul>
+        </section>
+        <!-- 8 slide -->
+        <section class="table-slide">
+          <h4 class="slide-name">TDD VS BDD</h4>
```
Comment (unfalse) [inline unit-testing/presentation.html:134]: Another bad practice in presentations: large amount of text with unreadable colors and tiny font size. Please make font size bigger and use small amount of words for each item. Write text for notes.

File: unit-testing/presentation.html [hunk 40]
```diff
@@ -2,43 +2,512 @@
...
+            file.​​​
+          </p>
+          <pre>
+          <code data-trim data-noescape data-line-numbers data-fragment-index="2">
+            module.exports = {
+              roots: ['&lt;rootDir>/src'],
+              collectCoverage: true,
+              coverageThreshold: { ... },
+              preset: 'react-native',
+              setupFiles: ['&lt;rootDir>jest.setup.js'],
```
Comment (unfalse) [inline unit-testing/presentation.html:289]: I guess it should be ['&lt;rootDir>/jest.setup.js']

File: unit-testing/presentation.html [hunk 41]
```diff
@@ -2,43 +2,512 @@
...
+            <pre
+              class="fragment"
+            ><code data-trim data-noescape data-line-numbers>
+          // Mock functions​​
+          jest.mock(moduleName, factory)​​
+          jest.unmock(moduleName)​​
+          jest.requireActual(moduleName)​​
+          jest.resetModules()​​
+          jest.isolateModules(fn)​​
+          // Mock modules​​
```
Comment (unfalse) [inline unit-testing/presentation.html:375]: Divide it into 2 fragments with code: Mock functions and Mock modules

File: unit-testing/presentation.html [hunk 42]
```diff
@@ -2,43 +2,512 @@
...
+        </code></pre>
+          </section>
+        </section>
+        <!-- 19 slide -->
+        <section class="text-and-conding-slide">
+          <h4 class="slide-name">Snapshots​​</h4>
+          <ul>
+            <li class="fragment">
+              Snapshot tests are used when you want to make sure your UI does
+              not change unexpectedly.
```
Comment (unfalse) [inline unit-testing/presentation.html:390]: Please write what snapshots  exactly are.

File: unit-testing/presentation.html [hunk 43]
```diff
@@ -2,43 +2,512 @@
...
+                100% unit test coverage does not mean we had good tests, or even
+                that the tests are complete.
+              </li>
+              <li class="fragment">
+                The tests could be missing important data and only testing with
+                data that succeeds, failing to test data that causes failures.
+              </li>
+              <li class="fragment">
+                Merely executing the code without checking its functionality
+                still counts in the coverage metrics.
```
Comment (unfalse) [inline unit-testing/presentation.html:434]: This slide looks like an article.
It's better to move most the most text into the notes and leave a simple list like
```
- 80% coverage is a good goal;
- higher coverage might be costly and doesn't give enough benefit
- 100% coverage doesn't mean good tests or that tests are complete
- (I don't quite understood it but it must be something like: "tests could be missing important data and test with data that succeeds and failing to test data that causes failures". What did you mean?)
- coverage metrics includes even the code that isn't checked for functionality
```

File: unit-testing/presentation.html [hunk 44]
```diff
@@ -2,43 +2,512 @@
...
+          <section class="picture-slide">
+            <h4 class="slide-name">Covarage report​</h4>
+            <img src="assets/images/coverage_1.png" alt="simple" />
+          </section>
+          <!-- 20/2 slide -->
+          <section class="picture-slide">
+            <h4 class="slide-name">
+              Covarage report in coverage/lcov-report/index.html​
+            </h4>
+            <img src="assets/images/coverage_2.png" alt="simple" />
```
Comment (unfalse) [inline unit-testing/presentation.html:448]: I think there should be text in notes that explains what values in columns are meaning.

File: unit-testing/presentation.html [hunk 45]
```diff
@@ -2,43 +2,512 @@
...
+              <li class="fragment">
+                Merely executing the code without checking its functionality
+                still counts in the coverage metrics.
+              </li>
+            </ul>
+          </section>
+          <!-- 20/1 slide -->
+          <section class="picture-slide">
+            <h4 class="slide-name">Covarage report​</h4>
+            <img src="assets/images/coverage_1.png" alt="simple" />
```
Comment (unfalse) [inline unit-testing/presentation.html:441]: I think there should be text in notes that explains what values in columns are meaning.

File: unit-testing/presentation.html [hunk 46]
```diff
@@ -2,43 +2,512 @@
...
+              </li>
+              <li class="fragment">
+                Merely executing the code without checking its functionality
+                still counts in the coverage metrics.
+              </li>
+            </ul>
+          </section>
+          <!-- 20/1 slide -->
+          <section class="picture-slide">
+            <h4 class="slide-name">Covarage report​</h4>
```
Comment (unfalse) [inline unit-testing/presentation.html:440]: Covarage -> Coverage

File: unit-testing/presentation.html [hunk 47]
```diff
@@ -2,43 +2,512 @@
...
+          <section class="picture-slide">
+            <h4 class="slide-name">
+              Covarage report in coverage/lcov-report/index.html​
+            </h4>
+            <img src="assets/images/coverage_2.png" alt="simple" />
+          </section>
+          <!-- 20/3 slide -->
+          <section class="picture-slide">
+            <h4 class="slide-name">Statement not covered​</h4>
+            <img src="assets/images/coverage_3.png" alt="simple" />
```
Comment (unfalse) [inline unit-testing/presentation.html:453]: add text about what does it mean

File: unit-testing/presentation.html [hunk 48]
```diff
@@ -2,43 +2,512 @@
...
+          </section>
+        </section>
+        <!-- 21 slide -->
+        <section class="picture-slide">
+          <h4 class="slide-name">When tests fail​​</h4>
+          <img src="assets/images/failed_tests.png" alt="simple" />
```
Comment (unfalse) [inline unit-testing/presentation.html:459]: add text into notes about what happened here

**Comment by unfalse** [comment]
I cannot approve this presentation as it needs a lot of work.

**Pull request merged**